src/views/charts/TradingRewardsChart.tsx

```diff
@@ -0,0 +1,261 @@
+import { STRING_KEYS } from '@/constants/localization';
+import { TOKEN_DECIMALS } from '@/constants/numbers';
+import { timeUnits } from '@/constants/time';
+
+import { useEnvConfig } from '@/hooks/useEnvConfig';
+import { useNow } from '@/hooks/useNow';
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useTokenConfigs } from '@/hooks/useTokenConfigs';
+
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AssetIcon } from '@/components/AssetIcon';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { TimeSeriesChart } from '@/components/visx/TimeSeriesChart';
+
+import { calculateCanViewAccount } from '@/state/accountCalculators';
+import {
+  getHistoricalTradingRewardsForPeriod,
+  getTotalTradingRewards,
+} from '@/state/accountSelectors';
+
+import abacusStateManager from '@/lib/abacus';
+import { formatRelativeTime } from '@/lib/dateTime';
+import { MustBigNumber } from '@/lib/numbers';
+
+type ElementProps = {
+  selectedLocale: string;
+  slotEmpty: React.ReactNode;
+};
+
+type StyleProps = {
+  className?: string;
+};
+
+const TRADING_REWARDS_TIME_RESOLUTION = 1 * timeUnits.hour;
+const SELECTED_PERIOD = HistoricalTradingRewardsPeriod.DAILY;
+
+const CHART_STYLES = {
+  margin: { left: 32, right: 48, top: 12, bottom: 32 },
+  padding: {
+    left: 0.01,
+    right: 0.01,
+    top: 0.5,
+    bottom: 0.01,
+  },
+};
+
+export const TradingRewardsChart = ({
+  selectedLocale,
+  slotEmpty,
+  className,
+}: ElementProps & StyleProps) => {
+  const stringGetter = useStringGetter();
+  const { chainTokenLabel } = useTokenConfigs();
+
+  const rewardsHistoryStartDate = useEnvConfig('rewardsHistoryStartDateMs');
+  const now = useNow({ intervalMs: timeUnits.minute });
+
+  const [tooltipContext, setTooltipContext] = useState<TooltipContextType<TradingRewardsDatum>>();
+  const [isZooming, setIsZooming] = useState(false);
+  const [selectedPeriod, setSelectedPeriod] = useState<TradingRewardsPeriod>(
+    TradingRewardsPeriod.PeriodAllTime
+  );
+
+  const canViewAccount = useSelector(calculateCanViewAccount);
+  const totalTradingRewards = useSelector(getTotalTradingRewards);
+  const periodTradingRewards = useSelector(
+    getHistoricalTradingRewardsForPeriod(SELECTED_PERIOD.name),
+    shallowEqual
+  );
+
+  useEffect(() => {
+    // Initialize daily data for rewards chart
+    abacusStateManager.setHistoricalTradingRewardPeriod(HistoricalTradingRewardsPeriod.DAILY);
+  }, [canViewAccount]);
+
+  const msForPeriod = useCallback(
+    (period: TradingRewardsPeriod) => {
+      switch (period) {
+        case TradingRewardsPeriod.Period1d:
+          return 1 * timeUnits.day;
+        case TradingRewardsPeriod.Period7d:
+          return 7 * timeUnits.day;
+        case TradingRewardsPeriod.Period30d:
+          return 30 * timeUnits.day;
+        case TradingRewardsPeriod.Period90d:
+          return 90 * timeUnits.day;
+        case TradingRewardsPeriod.PeriodAllTime:
```

Comment:
it seems like by default it should be all time, but for somehow it's unselected when it's mounted
<img width="918" alt="Screenshot 2024-05-28 at 4 25 45 PM" src="https://github.com/dydxprotocol/v4-web/assets/9400120/fd88711c-3c5f-46d4-b8ba-4f3b88f94d28">